lib/trackler/doc_file.rb

```diff
@@ -0,0 +1,92 @@
+class DocFile
+  DEFAULT_IMAGE_PATH = "/docs/img"
+
+  def self.find(basename:, track_dir:, image_path:)
+    dir = File.join(track_dir, "docs")
+
+    [
+      MarkdownFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      OrgmodeFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      NullFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+    ].detect(&:acceptable?)
+  end
+
+  attr_reader :basename, :dir, :image_path
+  def initialize(basename:, docs_dir:, image_path:)
+    @basename = basename
+    @dir = docs_dir
+    @image_path = image_path
+  end
+
+  def to_md
+    transform(body).gsub(img_src, img_dst)
+  end
+
+  def name
+    "%s.%s" % [basename, extension]
+  end
+
+  def extension
+    "md"
+  end
+
+  def acceptable?
+    File.exist?(path)
+  end
+
+  private
+
+  def body
+    File.read(path)
```

Comment:
I changed it to `to_s`, then I moved the image_file stuff directly to that method and changed the name to `render` since I don't like passing an argument (even an optional one) to `to_s`.